tests/lib/acorn-after-espree.js

```diff
@@ -19,11 +19,30 @@ const acorn = require("acorn"),
 
 describe("acorn", () => {
     it("acorn.parse() should work after espree was loaded.", () => {
-        const before = acorn.parse("var foo = bar /*world*/;");
+        const before = acorn.parse("var foo = bar /*world*/;", { ecmaVersion: 5 });
 
         require("../../espree");
-        const after = acorn.parse("var foo = bar /*world*/;");
+        const after = acorn.parse("var foo = bar /*world*/;", { ecmaVersion: 5 });
 
         assert.deepStrictEqual(after, before);
     });
+
+    it("acorn.parse() should not re-use the same identifier node in object patterns", () => {
+        const code = "({x, y = 1} = {z})";
+
+        const ast = acorn.parse(code, { ecmaVersion: 2015 });
```

Comment:
We should test what espree is producing at the end, not acorn directly.

If acorn reverts this change in the future for some reason, we could then clone the nodes in espree to retain the behavior.